### PR TITLE
ff cleanup: allow_votes_to_directly_update_vote_state and compact_vot…

### PR DESCRIPTION
**Pull request by AshwinSekar**
…e_state_updates

Activated mainnet 491

For:
#24717 
#26617 

Fixes #

### CODE REVIEW

File: programs/vote/src/vote_processor.rs [hunk 1]
```diff
@@ -180,48 +180,31 @@ declare_process_instruction!(
             }
             VoteInstruction::UpdateVoteState(vote_state_update)
```
Comment (AshwinSekar) [inline programs/vote/src/vote_processor.rs:181]: next step would be to remove this vote instruction, all votes should be coming over the wire compressed. realized I should've probably had a `!is_active(compact_vote_state_updates)` here.

**Comment by AshwinSekar** [comment]
Added @apfitzge for the banking_stage change.

**Review by carllin — APPROVED** [review]

File: core/src/consensus.rs [hunk 2]
```diff
@@ -620,7 +570,7 @@ impl Tower {
 
     #[cfg(test)]
     pub fn record_vote(&mut self, slot: Slot, hash: Hash) -> Option<Slot> {
```
Comment (apfitzge) [inline core/src/consensus.rs:572]: follow-up: this function seems unecessary now

**Review by apfitzge — APPROVED** [review]
LGTM. 
Minor comment for follow-up.
Feature is active everywhere:
```
$ sol_feature_status 86HpNqzutEZwLcPxS6EHDcMNYWk6ikhteg9un7Y2PBKE
          Feature                                      | Status                  | Activation Slot | Description
devnet:   86HpNqzutEZwLcPxS6EHDcMNYWk6ikhteg9un7Y2PBKE | active since epoch 437 | 188784000 | Compact vote state updates to lower block size
testnet:  86HpNqzutEZwLcPxS6EHDcMNYWk6ikhteg9un7Y2PBKE | active since epoch 516 | 217388260 | Compact vote state updates to lower block size
mainnet:  86HpNqzutEZwLcPxS6EHDcMNYWk6ikhteg9un7Y2PBKE | active since epoch 491 | 212112000 | Compact vote state updates to lower block size
```

**Pull request merged**